op-batcher/batcher/driver.go

```diff
@@ -867,9 +873,34 @@ func (l *BatchSubmitter) sendTx(txdata txData, isCancel bool, candidate *txmgr.T
 		candidate.GasLimit = intrinsicGas
 	}
 
+	floorDataGas, err := floorDataGas(candidate.TxData)
+	if err == nil && floorDataGas > candidate.GasLimit {
+		candidate.GasLimit = floorDataGas
+	}
+
 	queue.Send(txRef{id: txdata.ID(), isCancel: isCancel, isBlob: txdata.asBlob}, *candidate, receiptsCh)
 }
 
+// Copypaste from upstream geth
+// TODO replace with import once it's available https://github.com/ethereum-optimism/op-geth/pull/507
```

Comment:
```suggestion
// TODO remove in #14500
```
I'll do it in my upstream merge.